src/spell.c

```diff
@@ -405,6 +405,36 @@ static linenr_T dump_prefixes(slang_T *slang, char_u *word, char_u *pat, int *di
 static char_u	*repl_from = NULL;
 static char_u	*repl_to = NULL;
 
+static void
+ensurehunspellinit(lp)
+    slang_T *lp;
+{
+    if (!lp->sl_hunspell)
+    {
+        char_u *dic = lp->sl_fname;
+        char_u *aff = vim_strnsave(dic, strlen(dic));
+
+	vim_strncpy(aff + strlen(aff) - 3, "aff", 3);
+
+	lp->sl_hunspell = Hunspell_create(aff, dic);
```

Comment:
All the calls to the hunspell API are unguarded, so they're being used even when the code wasn't configured to use hunspell.  The commit I linked earlier defines `FEAT_HUNSPELL` which can be used for `#ifdef` checks to disable hunspell-specific code when it isn't needed.

Reply:
Ehm, I don't see ``FEAT_HUNSPELL`` in https://github.com/jamessan/vim/commit/09f1bf8d4c721f1b4fe50c22a345d4cf5129e7a3 , what I do wrong? Any pull requests against my branch https://github.com/mcepl/vim/tree/hunspell are more than welcome, of course.

Reply:
It's in https://github.com/jamessan/vim/commit/4c89bbe0bbaa5772b23c2882c7036728db90fa3e but I hadn't gotten around to the C side of things yet.  I was breaking down the patch into logical commits as I tried to reason through the changes that the patch made.

I didn't finish the branch at the time because I took a detour through implementing [support for enchant](https://github.com/jamessan/vim/tree/pu/enchant-fallback) instead.  I abandoned that eventually because enchant's APIs don't integrate as well with Vim's heuristics and just never circled back to hunspell (which does more closely align with Vim's existing code).

Reply:
I was contemplating enchant as well, but as they say “"We can solve any problem by introducing an extra level of indirection except for the problem of too many levels of indirection.” With Open/LibreOffice we can hope for wide support of ``hunspell`` on many platforms, but ``enchant`` requires supporting yet another library on all those platforms for questionable benefit, so if I am not missing anything, I would rather stay with supporting ``hunspell`` directly.

Perhaps, though, we should cover all this patch with enough ``#ifdef FEAT_HUNSPELL`` and protect the original code, so that we at least preserve backwards compatibility.